sources/packages/backend/apps/api/src/services/cas-invoice-batch/cas-invoice-batch.service.ts

```diff
@@ -47,4 +56,64 @@ export class CASInvoiceBatchService {
     });
     return { results, count };
   }
+
+  /**
+   * Update the approval status for a CAS invoice batch record.
+   * @param casInvoiceBatchId ID of the CAS invoice batch to be updated.
+   * @param approvalStatus approval status.
+   * @param auditUserId user to update the record.
+   */
+  async updateCASInvoiceBatch(
+    casInvoiceBatchId: number,
+    approvalStatus: CASInvoiceBatchApprovalStatus,
+    auditUserId: number,
+  ): Promise<void> {
+    const casInvoiceBatch = await this.getCASInvoiceBatchRecord(
+      casInvoiceBatchId,
+    );
+    if (!casInvoiceBatch) {
+      throw new CustomNamedError(
+        `CAS invoice batch with ID ${casInvoiceBatchId} not found.`,
+        CAS_INVOICE_BATCH_NOT_FOUND,
+      );
+    }
+    if (
+      casInvoiceBatch.approvalStatus !== CASInvoiceBatchApprovalStatus.Pending
+    ) {
+      throw new CustomNamedError(
+        `Cannot update CAS invoice batch with ID ${casInvoiceBatchId} that is approved or rejected.`,
+        CAS_INVOICE_BATCH_NOT_PENDING,
+      );
+    }
+    const auditUser = { id: auditUserId } as User;
+    try {
+      await this.casInvoiceBatchRepo.update(
+        { id: casInvoiceBatchId },
+        {
+          approvalStatus,
+          approvalStatusUpdatedBy: auditUser,
+          approvalStatusUpdatedOn: new Date(),
+        },
+      );
+    } catch (error: unknown) {
+      throw error;
+    }
```

Comment:
Please remove the try/catch since no additional operation is performed and the result of having it or not would be the same.